### PR TITLE
DataGrid, TreeList - Should be possible navigate between views using Ctrl+ArrowUp, Ctrl+ArrowDown keys

### PR DESCRIPTION
**Pull request by AntonSermyazhko**

### CODE REVIEW

File: styles/widgets/base/treeList.less [hunk 1]
```diff
@@ -49,6 +49,7 @@
         .dx-icon-sortdown;
     }
 
+
```
Comment (vconst) [inline styles/widgets/base/treeList.less]: Why?

File: testing/helpers/elements/data_grid/elementBase.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import $ from "jquery";
+
+export class elementBase {
```
Comment (vconst) [inline testing/helpers/elements/data_grid/elementBase.js]: elementBase -> ElementBase

File: testing/helpers/elements/data_grid/filterPanelElement.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { elementBase } from './elementBase.js';
+
+export class filterPanelElement extends elementBase {
```
Comment (vconst) [inline testing/helpers/elements/data_grid/filterPanelElement.js]: filterPanelElement -> FilterPanelElement

File: testing/helpers/elements/data_grid/pagerElement.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { elementBase } from './elementBase.js';
+
+export class pagerElement extends elementBase {
```
Comment (vconst) [inline testing/helpers/elements/data_grid/pagerElement.js]: pagerElement -> PagerElement

File: testing/helpers/elements/data_grid/filterPanelElement.js (hunk 3 above)
Comment (vconst) [reply testing/helpers/elements/data_grid/filterPanelElement.js]: May be use FilterPanelWrapper name? and BaseWrapper?

File: js/ui/grid_core/ui.grid_core.column_headers.js [hunk 5]
```diff
@@ -176,7 +176,7 @@ module.exports = {
                     if(row.rowType === "header") {
                         $row.addClass(HEADER_ROW_CLASS);
                         if(!this._isLegacyKeyboardNavigation()) {
-                            registerKeyboardAction(this, $row, "td", this._handleActionKeyDown.bind(this));
+                            registerKeyboardAction("headerColumns", this, $row, "td", this._handleActionKeyDown.bind(this));
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.column_headers.js]: Rename headerColumns to columnHeaders for consistency with view name

File: js/ui/pager.js [hunk 6]
```diff
@@ -14,7 +14,8 @@ var $ = require("../core/renderer"),
     SelectBox = require("./select_box"),
     NumberBox = require("./number_box"),
     eventUtils = require("../events/utils"),
-    registerKeyboardAction = require("./shared/accessibility").registerKeyboardAction;
+    registerKeyboardAction = require("./shared/accessibility").registerKeyboardAction,
+    setTabIndex = require("./shared/accessibility").setTabIndex;
```
Comment (vconst) [inline js/ui/pager.js]: require("./shared/accessibility") is duplicated. Extract module to variable

File: js/ui/shared/accessibility.js [hunk 7]
```diff
@@ -2,18 +2,48 @@ import $ from "../../core/renderer";
...
+}
+
+function findFocusedViewElement(instance, viewSelectors) {
+    for(let index in viewSelectors) {
+        let componentName = instance.component && instance.component.NAME,
+            selector = viewSelectors[index],
+            $focusViewElement;
+
+        if(componentName === "dxTreeList") {
+            selector = selector.replace(/dx-datagrid/g, "dx-treelist");
```
Comment (vconst) [inline js/ui/shared/accessibility.js]: Very strange code.

File: js/ui/shared/accessibility.js [hunk 8]
```diff
@@ -2,18 +2,48 @@ import $ from "../../core/renderer";
...
 import eventUtils from "../../events/utils";
 
-const FOCUS_STATE_CLASS = "dx-state-focused";
+const FOCUS_STATE_CLASS = "dx-state-focused",
+    FOCUS_DISABLED_CLASS = "dx-cell-focus-disabled",
+    DATA_ROW_SELECTOR = ".dx-datagrid-rowsview .dx-datagrid-content .dx-data-row",
+    viewItemSelectorMap = {
+        headerColumns: [".dx-datagrid-headers .dx-header-row > td.dx-datagrid-action"],
+        filterRow: [".dx-datagrid-headers .dx-datagrid-filter-row .dx-editor-cell input"],
+        rowsView: [`${DATA_ROW_SELECTOR} > td[tabindex]`, `${DATA_ROW_SELECTOR} > td`],
```
Comment (vconst) [inline js/ui/shared/accessibility.js]: This selected is wrong for grouped DataGrid

File: js/ui/shared/accessibility.js [hunk 9]
```diff
@@ -2,18 +2,48 @@ import $ from "../../core/renderer";
...
+        footer: [".dx-datagrid-total-footer .dx-datagrid-summary-item"],
+        filterPanel: [".dx-datagrid-filter-panel .dx-icon-filter"],
+        pager: [".dx-datagrid-pager [tabindex]", ".dx-datagrid-pager .dx-selection"]
```
Comment (vconst) [inline js/ui/shared/accessibility.js]: Is ".dx-datagrid-pager .dx-selection" selector necessary?

File: js/ui/shared/accessibility.js (hunk 7 above)
Comment (vconst) [reply js/ui/shared/accessibility.js]: datagrid and treelist names should not be in this file

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/ui/shared/accessibility.js (hunk 8 above)
Comment (AntonSermyazhko) [reply js/ui/shared/accessibility.js]: Fixed

File: js/ui/shared/accessibility.js (hunk 9 above)
Comment (AntonSermyazhko) [reply js/ui/shared/accessibility.js]: Fixed

**Review by vconst — APPROVED** [review]

**Pull request merged**